packages/test_track/lib/src/networking/interceptors/retry_interceptor.dart

```diff
@@ -1,31 +1,42 @@
 import 'dart:async';
 
 import 'package:dio/dio.dart';
-import 'package:test_track/src/networking/http_client.dart';
+import 'package:sturdy_http/sturdy_http.dart';
+import 'package:test_track/src/networking/extras.dart';
 import 'package:test_track/src/networking/interceptors/retry_options.dart';
 
+/// Function that must return the [SturdyHttp] instance that will
+/// handle the retry.
+typedef SturdyHttpGetter = SturdyHttp Function();
+
 /// {@template retry_interceptor}
 /// An [Interceptor] responsible for performing retry
 /// operations in the case of a failed network request
 /// {@endtemplate}
 class RetryInterceptor extends Interceptor {
-  final Dio _dio;
+  final SturdyHttpGetter _clientGetter;
   final RetryOptions _retryOptions;
 
   /// {@macro retry_interceptor}
   RetryInterceptor({
-    required Dio dio,
+    required SturdyHttpGetter clientGetter,
     RetryOptions? retryOptions,
-  })  : _dio = dio,
+  })  : _clientGetter = clientGetter,
         _retryOptions = retryOptions ?? RetryOptions();
 
   @override
-  Future onError(DioError err, ErrorInterceptorHandler handler) async {
+  Future<void> onError(
+      DioException err, ErrorInterceptorHandler handler) async {
     var extra =
         RetryOptions.fromRequestOptions(err.requestOptions) ?? _retryOptions;
     final isIdempotent =
-        err.requestOptions.extra[HttpClient.isIdempotentOptionsKey] as bool? ??
-            false;
+        err.requestOptions.extra[isIdempotentOptionsKey] as bool? ?? false;
+    final uppercased = StringBuffer()
```

Comment:
we should probably call this pascalCased because it's not quite UPPERCASED it's more PascalCased, right?

Reply:
or in simpler terms, we could called it `capitalized` as in `Capitalized`

Reply:
Hehe yeah